### PR TITLE
Fix propagation of Kafka key for messages not generated by Silverback…

### PR DESCRIPTION
**Pull request by BEagle1984**
This is related to the issue #60.
What I do here is instead of relying on the header, I properly read the Kafka key from the client library and I set the header in the consumer.

### CODE REVIEW

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs [hunk 1]
```diff
@@ -78,15 +76,20 @@ protected override async Task<IOffset> ProduceAsync(RawBrokerEnvelope envelope)
             }
         }
 
-        private byte[] GetPartitioningKey(IEnumerable<MessageHeader> headers)
+        private byte[] GetKafkaMessageKey(MessageHeaderCollection headers)
```
Comment (msallin) [inline src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs]: Rename method to reflect that the header is removed from the collection, if present.

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaBroker.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@ namespace Silverback.Messaging.Broker
     /// </summary>
     public class KafkaBroker : Broker
     {
+        internal const string MessageKeyHeaderKey = "x-kafka-message-key";
```
Comment (msallin) [inline src/Silverback.Integration.Kafka/Messaging/Broker/KafkaBroker.cs]: Would it make sense to put this const somewhere else and make it public? According to the issue the intention is to use it from outside, isn't it?

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs (hunk 1 above)
Comment (BEagle1984) [reply src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs]: :) I like that you have my same mindset when it goes to self documenting code. I will definitely change it (I couldn't come up with a nice name and I left it alone).

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaBroker.cs (hunk 2 above)
Comment (BEagle1984) [reply src/Silverback.Integration.Kafka/Messaging/Broker/KafkaBroker.cs]: Oops, yes, that should definitely be public now. Thanks!

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs (hunk 1 above)
Comment (BEagle1984) [reply src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs]: Name changed!

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaBroker.cs (hunk 2 above)
Comment (BEagle1984) [reply src/Silverback.Integration.Kafka/Messaging/Broker/KafkaBroker.cs]: Moved all constants into a separate class (public of course).

**Pull request merged**